src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -280,12 +289,21 @@ class LocalLambdaRunner {
             this.localInvokeArgs.handlerName
         ).replace('\\', '/')
 
-        await new SamTemplateGenerator()
+        const existingTemplateResource =
+            await this.getResourceForHandlerFromTemplate(relativeFunctionHandler, await detectLocalLambdas())
+
+        const newTemplate = new SamTemplateGenerator()
             .withCodeUri(rootCodeFolder)
             .withFunctionHandler(relativeFunctionHandler)
             .withResourceName(LocalLambdaRunner.TEMPLATE_RESOURCE_NAME)
             .withRuntime(this.runtime)
-            .generate(inputTemplatePath)
+
+        if (!!existingTemplateResource && !!existingTemplateResource.Properties &&
+            existingTemplateResource.Properties.Environment) {
+            newTemplate.withEnvironment(existingTemplateResource.Properties.Environment)
```

Comment:
Nit: The fact that `withEnvironment` returns `this` rather than a new object is an implementation detail of `SamTemplateGenerator`. Please change to `newTemplate = newTemplate.withEnvironment` to protect against changes to `SamTemplateGenerator` (this also requires changing `const newTemplate` to `let newTemplate`).